### DISCUSSION TITLE
Is <Link> always expected to call loader() via fetch from the client?

### DISCUSSION THREAD

**Discussion by davnicwil** [remix-run/remix#943] [Q&A] [answered]
In the docs [here](https://remix.run/docs/en/v1/guides/api-routes#routes-are-their-own-api) it mentions

> Whenever the user clicks a link to `<Link to="/teams" />`, Remix in the browser will perform the fetch to the server to get the data from the loader and render the route.

This is what I'd expect - remix doing client side nav and fetching the data as opposed to a full server render like you'd get with an `<a>` tag. 

But I've noticed when adding random pages to the demo app (from `create-remix-app`) it in fact does seem to fully reload the page for `<Link>`s to those pages. The only place it seems to do the above is for nested routes using `<Outlet>` - ie just does client side nav and loads just the data from the nested route component's loader via fetch.

Is this the expected behaviour?

**Comment by ryanflorence** [comment, marked as answer]
Remix knows which route layouts will change in the UI and only calls the loaders for changing routes.

**Reply by sergiodxa** [reply]
From what I understood, the issue is that the Link is not doing a client-side navigation for some routes.

If that's the case, @davnicwil could you create a reproduction in [https://remix.new](https://remix.new)?

**Reply by davnicwil** [reply]
@sergiodxa you're right, but I can't now reproduce this issue -- I was definitely seeing this a couple of weeks ago just playing around with Remix for the first time, with the default demo app. The only reason I raised the question was this behaviour seeming so unexpected. But anyway, works exactly as expected now, calling the loader via fetch on client side nav on every route (including 'base' ones). 

As a complete guess here, this might be a genuine bug if some error is thrown _somewhere_ and this causes Remix to do a full reload instead of showing the error, something like that. Again, this was just the default demo app with some hacky stuff thrown on top to play around, so very likely something somewhere was throwing errors!

Sorry can't be more helpful, if I do manage to reproduce by chance at some point I'll post an example!
